### ISSUE TITLE
Support for GR00T-N1.5-3B Model in IsaacLabEvalTasks

### ISSUE THREAD

**Issue by SahasraKeerthi** [isaac-sim/IsaacLabEvalTasks#34]
# IsaacLabEvalTasks Limitations

```
Expand IsaacLabEvalTasks to support GR00T-N1.5-3B and Generic Task Evaluation
```

### Problem 1: Limited Model Version Support

IsaacLabEvalTasks only supports GR00T-N1-2B checkpoints. GR00T-N1.5-3B (latest model) is incompatible with the evaluation pipeline.

**Current Behavior:**
```
Error: Model configuration mismatch - Expected N1-2B, got N1.5-3B architecture
```

**Expected Behavior:**
```bash
python scripts/play.py \
  --checkpoint path/to/GR00T-N1.5-3B-checkpoint.pt
```

### Problem 2: Only Two Hardcoded Tasks

IsaacLabEvalTasks only provides two pre-built tasks:
- `Isaac-Groot-GR1-NutPouring-v0`
- `Isaac-Groot-GR1-ExhaustPipeSorting-v0`

Users cannot evaluate on custom tasks or different task types without modifying the core codebase. This severely limits the framework's usability.

**Current Limitation:**
- Cannot evaluate on user-defined manipulation tasks
- Cannot test models on different robot configurations
- Cannot benchmark on custom object manipulation scenarios
- Requires deep understanding of codebase to add new tasks

**Expected Behavior:**

Users should be able to:
1. Define custom tasks in configuration files
2. Load and evaluate any pretrained or fine-tuned GR00T model
3. Run inference on arbitrary manipulation tasks
4. Generate evaluation metrics without code modification

---

## Root Causes

**Issue 1 - Model Version:**
- Model loading expects N1-2B checkpoint structure
- Embodiment tag mappings not updated for N1.5
- Data config incompatible with N1.5 architecture

**Issue 2 - Task Limitation:**
- Tasks hardcoded into evaluation script
- No task abstraction or plugin system
- No way to specify custom task parameters
- Limited to environments defined in repository

---

### Solution 1: Add Model Version Detection

```python
def detect_model_version(checkpoint_path):
    """Auto-detect GR00T model version"""
    checkpoint = torch.load(checkpoint_path)
    model_config = checkpoint.get("model_config", {})
    
    # Support N1-2B, N1.5-3B, and future versions
    if model_config.get("num_parameters") == 3e9:
        return "N1.5-3B"
    elif model_config.get("num_parameters") == 2e9:
        return "N1-2B"
    else:
        raise ValueError("Unknown model version")
```

### Solution 2: Generic Task System

Allow users to specify tasks via configuration:

```bash
# Evaluate pretrained model on custom task
python scripts/play.py \
  --task-config tasks/my_custom_task.yaml \
  --checkpoint path/to/model.pt \
  --num_envs 4

# Or with task parameters
python scripts/play.py \
  --task-name "pick_and_place" \
  --task-params '{"object": "cube", "target": "bin"}' \
  --checkpoint path/to/model.pt
```

**Task Configuration File (`tasks/my_custom_task.yaml`):**
```yaml
task_name: pick_and_place
environment: Isaac-Groot-GR1-PickPlace-v0
robot: gr1_humanoid
duration: 500
success_metric: object_in_target
visualization: true
```

### Solution 3: Support Generic Model Loading

```python
# Load any GR00T model variant
policy = load_groot_model(
    checkpoint_path="path/to/checkpoint.pt",
    auto_detect_version=True  # Automatically detect N1-2B or N1.5-3B
)

# Or explicit version
policy = load_groot_model(
    checkpoint_path="path/to/checkpoint.pt",
    model_version="N1.5-3B"
)
```

---

## Impact

**Current Limitations Prevent:**
- Evaluating latest GR00T-N1.5-3B models
- Custom task evaluation
- Benchmarking on user-specific scenarios
- Easy integration into research workflows
- Extension by community without core modifications

**After Fix - Users Can:**
- Evaluate any GR00T model version (pre-trained or fine-tuned)
- Define custom tasks in YAML/JSON
- Benchmark on arbitrary manipulation scenarios
- Add new tasks without code changes
- Use framework for any GR00T-based research

---

## Implementation Steps

- Add model version detection and support for N1.5-3B
- Create task abstraction/plugin system
- Allow task definition via configuration files
- Support dynamic task loading
- Document task creation guide
- Provide examples of custom tasks
- Add backward compatibility for existing tasks
- Update evaluation script to use new system

---

## Affected Users

- Researchers evaluating latest GR00T models
- Teams fine-tuning models on custom tasks
- Anyone wanting to benchmark on non-standard tasks
- Community members extending the framework

---

## Labels

`feature-request` `GR00T` `evaluation` `compatibility` `extensibility` `task-system`

**Comment by xyao-nv** [comment]
Thanks for the detailed suggestion! Indeed, we are planning to deprecate this repo as we have released [IsaacLab-Arena](https://github.com/isaac-sim/IsaacLab-Arena), a scalable environment creation and evaluation framework for robotics simulations. With this easy-to-use framework, it enables more tasks and comes with N1.5 supports. Please check out there and see if it fits your need.
